### PR TITLE
Feature lpdev 684 update log system

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Service.PayInternal.Services/CalculationService.cs [hunk 1]
```diff
@@ -38,15 +38,15 @@ public class CalculationService : ICalculationService
         {
             var rate = await GetRateAsync(baseAssetId, quotingAssetId, requestMarkup.Percent, requestMarkup.Pips, merchantMarkup);
 
-            await _log.WriteInfoAsync(nameof(CalculationService), nameof(GetAmountAsync), new
+            _log.Info("Rate calculation", new
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/CalculationService.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal.Services/CalculationService.cs [hunk 2]
```diff
@@ -118,8 +116,11 @@ public class CalculationService : ICalculationService
                 }
             }
 
-            await _log.WriteInfoAsync(nameof(CalculationService), nameof(GetRateAsync),
-                new {askPrice, bidPrice}.ToJson(), "Market rate that will be used for calculation");
+            _log.Info("Market rate that will be used for calculation", new
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/CalculationService.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal.Services/CalculationService.cs [hunk 3]
```diff
@@ -159,8 +160,11 @@ public class CalculationService : ICalculationService
             PriceCalculationMethod priceValueType,
             IMarkup merchantMarkup)
         {
-            _log.WriteInfoAsync(nameof(CalculationService), nameof(CalculatePrice), new {askPrice, bidPrice}.ToJson(),
-                "Rate calculation").GetAwaiter().GetResult();
+            _log.Info("Rate calculation", new
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/CalculationService.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal.Services/EthereumIataApiClient.cs [hunk 4]
```diff
@@ -133,8 +134,7 @@ public class EthereumIataApiClient : IBlockchainApiClient
 
             if (response is ApiException ex)
             {
-                _log.WriteWarning(nameof(CreateAddressAsync), "New erc20 address generation",
-                    ex.Error?.Message);
+                _log.Warning("New erc20 address generation", context: ex);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/EthereumIataApiClient.cs]: Why exception is put in context insead of exception parameter?

File: src/Lykke.Service.PayInternal.Services/ShutdownManager.cs [hunk 5]
```diff
@@ -35,11 +35,9 @@ public ShutdownManager(ILog log, IEnumerable<IStopable> items)
                 }
                 catch (Exception ex)
                 {
-                    _log.WriteWarning(nameof(StopAsync), null, $"Unable to stop {item.GetType().Name}", ex);
+                    _log.Warning(nameof(Stop), $"Unable to stop {item.GetType().Name}", ex);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/ShutdownManager.cs]: nameof(Stop) is redundant

File: src/Lykke.Service.PayInternal.Services/StartupManager.cs [hunk 6]
```diff
@@ -19,33 +19,33 @@ namespace Lykke.Service.PayInternal.Services
...
             _paymentRequestPublisher = paymentRequestPublisher ?? throw new ArgumentNullException(nameof(paymentRequestPublisher));
             _transactionPublisher = transactionPublisher ?? throw new ArgumentNullException(nameof(transactionPublisher));
+            _log = logFactory.CreateLog(this);
         }
 
-        public async Task StartAsync()
+        public void Start()
         {
-            await _log.WriteInfoAsync(nameof(StartAsync), string.Empty, "Checking app settings consistency...");
+            _log.Info(nameof(Start), "Checking app settings consistency...");
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/StartupManager.cs]: nameof(Start) is redundant

File: src/Lykke.Service.PayInternal.Services/StartupManager.cs [hunk 7]
```diff
@@ -54,30 +54,30 @@ public class StartupManager : IStartupManager
             if (primaryExpPeriod > extendedExpPeriod)
                 throw new OrderExpirationSettingsInconsistentException(primaryExpPeriod, extendedExpPeriod);
 
-            await _log.WriteInfoAsync(nameof(StartAsync), string.Empty, "Settings checked successfully.");
+            _log.Info(nameof(Start), "Settings checked successfully.");
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/StartupManager.cs]: nameof(Start) is redundant

File: src/Lykke.Service.PayInternal.Services/StartupManager.cs [hunk 8]
```diff
@@ -54,30 +54,30 @@ public class StartupManager : IStartupManager
...
 
-            StartComponentAsync("Transaction publisher", _transactionPublisher);
+            StartComponent("Transaction publisher", _transactionPublisher);
         }
 
-        private void StartComponentAsync(string componentDisplayName, object component)
+        private void StartComponent(string componentDisplayName, object component)
         {
-            _log.WriteInfo(nameof(StartAsync), string.Empty, $"Starting {componentDisplayName} ...");
+            _log.Info(nameof(Start), $"Starting {componentDisplayName} ...");
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/StartupManager.cs]: nameof(Start) is redundant

File: src/Lykke.Service.PayInternal.Services/StartupManager.cs [hunk 9]
```diff
@@ -54,30 +54,30 @@ public class StartupManager : IStartupManager
...
 
             if (component is IStartable startableComponent)
             {
                 startableComponent.Start();
 
-                _log.WriteInfo(nameof(StartAsync), string.Empty, $"{componentDisplayName} successfully started.");
+                _log.Info(nameof(Start), $"{componentDisplayName} successfully started.");
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/StartupManager.cs]: nameof(Start) is redundant

File: src/Lykke.Service.PayInternal.Services/StartupManager.cs [hunk 10]
```diff
@@ -54,30 +54,30 @@ public class StartupManager : IStartupManager
...
             }
             else
             {
-                _log.WriteWarning(nameof(StartAsync), component, "Component has not been started");
+                _log.Warning(nameof(Start), "Component has not been started", context: component);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal.Services/StartupManager.cs]: nameof(Start) is redundant

File: src/Lykke.Service.PayInternal/Controllers/MerchantWalletsController.cs [hunk 11]
```diff
@@ -73,19 +75,19 @@ public class MerchantWalletsController : Controller
             }
             catch (InvalidOperationException e)
             {
-                _log.WriteError(nameof(Create), request, e);
+                _log.Error(e, request);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal/Controllers/MerchantWalletsController.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal/Controllers/MerchantsController.cs [hunk 12]
```diff
@@ -173,24 +172,22 @@ public class MerchantsController : Controller
...
+                    e.Value
+                });
 
-                return BadRequest(ErrorResponse.Create(ex.Message));
+                return BadRequest(ErrorResponse.Create(e.Message));
             }
             catch (MerchantNotFoundException exception)
             {
-                _log.WriteWarning(nameof(UpdateAsync), request, exception.Message);
+                _log.Warning(exception.Message, context: request);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal/Controllers/MerchantsController.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal/Controllers/PaymentRequestsController.cs [hunk 13]
```diff
@@ -245,33 +224,31 @@ public class PaymentRequestsController : Controller
...
+            catch (AssetNetworkNotDefinedException e)
             {
-                await _log.WriteErrorAsync(nameof(RefundAsync), new {networkEx.AssetId}.ToJson(), networkEx);
+                _log.Error(e, new {e.AssetId});
             }
-            catch (Exception ex)
+            catch (Exception e)
             {
-                await _log.WriteErrorAsync(nameof(RefundAsync), request.ToJson(), ex);
+                _log.Error(e, request);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal/Controllers/PaymentRequestsController.cs]: Alexey Novicikhin usually suggests to save in context only data wich is used for searching by fields in kibana.

File: src/Lykke.Service.PayInternal/Startup.cs [hunk 14]
```diff
@@ -123,119 +134,63 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
...
                 ApplicationContainer.Dispose();
             }
             catch (Exception ex)
             {
-                if (Log != null)
-                {
-                    await Log.WriteFatalErrorAsync(nameof(Startup), nameof(CleanUp), "", ex);
-                    (Log as IDisposable)?.Dispose();
-                }
+                Log?.Critical(nameof(CleanUp), ex);
```
Comment (LatheLykke) [inline src/Lykke.Service.PayInternal/Startup.cs]: nameof(CleanUp) is redundant

File: src/Lykke.Service.PayInternal.Services/EthereumIataApiClient.cs (hunk 4 above)
Comment (tarurar) [reply src/Lykke.Service.PayInternal.Services/EthereumIataApiClient.cs]: It is not an exception. ApiException is a plain DTO class declared in EthereumCore client library.

**Pull request merged**